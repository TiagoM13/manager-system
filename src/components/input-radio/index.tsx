import React from 'react';
import { Control, Controller, FieldError } from 'react-hook-form';

interface InputRadioProps {
  required?: boolean;
  disabled?: boolean;
  label?: string;
  name?: string;
  options: {
    opt1: string;
    opt2: string;
  };
  error?: FieldError | undefined;
}

export const InputRadio: React.FC<InputRadioProps> = ({
  required = false,
  disabled = false,
  label = 'Selecione uma opção',
  options,
  name,
  error,
}) => {
  return (
    <>
      <label className="block text-sm text-slate-600 mb-2">
        {label} {required && <span className="text-red-500">*</span>}
      </label>

      <div className="flex gap-6">
        <OptionRadio
          name={name}
          id={options.opt1}
          value={options.opt1}
          disabled={disabled}
        />
        <OptionRadio
          name={name}
          id={options.opt2}
          value={options.opt2}
          disabled={disabled}
        />
      </div>

      {!!error && (
        <span className="block text-sm text-red-500 mt-1">{error.message}</span>
      )}
    </>
  );
};

type OptionRadioProps = React.ComponentProps<'input'> & {
  control?: Control<any>;
  name?: string;
};

const OptionRadio = ({ name, control, ...props }: OptionRadioProps) => {
  return (
    <div className="flex items-center gap-2.5">
      <Controller
        name={name as string}
        control={control}
        render={({ field }) => (
          <input
            {...field}
            {...props}
            id={props.id?.toLowerCase()}
            type="radio"
            className="cursor-pointer appearance-none h-[12px] w-[12px] border border-slate-600 rounded-full hover:bg-sky-200 checked:bg-sky-600 checked:hover:bg-sky-500 transition-all ease-in duration-500 disabled:opacity-60"
          />
        )}
      />

      <label
        htmlFor={props.id?.toLowerCase()}
        className={`cursor-pointer capitalize text-slate-600 ${props.disabled ? 'opacity-60' : null}`}
      >
        {props.value}
      </label>
    </div>
  );
};
