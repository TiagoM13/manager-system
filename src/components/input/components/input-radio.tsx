import React from 'react';
import { Control, Controller } from 'react-hook-form';

import { ErrorMessage } from '@/components/error-message';

import { InputRadioProps } from './interfaces';

export const InputRadio: React.FC<InputRadioProps> = ({
  required = false,
  disabled = false,
  label = 'Selecione uma opção',
  control,
  options,
  name,
  error,
}) => {
  return (
    <>
      <label className="block text-sm text-slate-600 mb-2">
        {label} {required && <span className="text-red-500">*</span>}
      </label>

      <div className="flex gap-6">
        <OptionRadio
          name={name}
          control={control}
          id={options.opt1}
          value={options.opt1}
          disabled={disabled}
        />
        <OptionRadio
          name={name}
          control={control}
          id={options.opt2}
          value={options.opt2}
          disabled={disabled}
        />
      </div>

      <ErrorMessage error={error?.message} />
    </>
  );
};

type OptionRadioProps = React.ComponentProps<'input'> & {
  control?: Control<any>;
  defaultValue?: any;
  name?: string;
};

const OptionRadio = ({
  name,
  control,
  defaultValue,
  ...props
}: OptionRadioProps) => {
  return (
    <div className="flex items-center gap-2.5">
      {control ? (
        <Controller
          name={name as string}
          control={control}
          defaultValue={defaultValue}
          render={({ field }) => (
            <input
              {...field}
              {...props}
              checked={field.value === props.value}
              id={props.id?.toLowerCase()}
              type="radio"
              className={`cursor-pointer appearance-none h-[12px] w-[12px] border border-slate-600 rounded-full  transition-all ease-in duration-500 disabled:opacity-60 disabled:cursor-auto checked:bg-sky-600 checked:hover:bg-sky-500 ${!props.disabled ? 'hover:bg-sky-200' : ''}`}
            />
          )}
        />
      ) : (
        <input
          {...props}
          id={props.id?.toLowerCase()}
          type="radio"
          className={`cursor-pointer appearance-none h-[12px] w-[12px] border border-slate-600 rounded-full  transition-all ease-in duration-500 disabled:opacity-60 disabled:cursor-auto checked:bg-sky-600 checked:hover:bg-sky-500 ${!props.disabled ? 'hover:bg-sky-200' : ''}`}
        />
      )}

      <label
        htmlFor={props.id?.toLowerCase()}
        className={`cursor-pointer capitalize text-slate-600 ${props.disabled ? 'opacity-60 cursor-auto' : ''}`}
      >
        {props.value}
      </label>
    </div>
  );
};
