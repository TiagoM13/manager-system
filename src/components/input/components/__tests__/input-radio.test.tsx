import React from 'react';

import { render } from '@testing-library/react';

import { InputRadio } from '../input-radio';

describe('<InputRadio/>', () => {
  test('should render correctly component', () => {
    const opts = {
      opt1: 'Sim',
      opt2: 'Não',
    };

    const label = 'Teste de seleção';

    const { getByText, asFragment } = render(
      <InputRadio label={label} options={opts} />,
    );

    expect(getByText(label)).toBeInTheDocument();
    expect(getByText(opts.opt1)).toBeInTheDocument();
    expect(getByText(opts.opt2)).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
