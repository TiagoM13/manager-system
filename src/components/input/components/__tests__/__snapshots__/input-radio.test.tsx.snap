// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputRadio/> should render correctly component 1`] = `
<DocumentFragment>
  <label
    class="block text-sm text-slate-600 mb-2"
  >
    Teste de seleção 
  </label>
  <div
    class="flex gap-6"
  >
    <div
      class="flex items-center gap-2.5"
    >
      <input
        class="cursor-pointer appearance-none h-[12px] w-[12px] border border-slate-600 rounded-full  transition-all ease-in duration-500 disabled:opacity-60 disabled:cursor-auto checked:bg-sky-600 checked:hover:bg-sky-500 hover:bg-sky-200"
        id="sim"
        type="radio"
        value="Sim"
      />
      <label
        class="cursor-pointer capitalize text-slate-600 "
        for="sim"
      >
        Sim
      </label>
    </div>
    <div
      class="flex items-center gap-2.5"
    >
      <input
        class="cursor-pointer appearance-none h-[12px] w-[12px] border border-slate-600 rounded-full  transition-all ease-in duration-500 disabled:opacity-60 disabled:cursor-auto checked:bg-sky-600 checked:hover:bg-sky-500 hover:bg-sky-200"
        id="não"
        type="radio"
        value="Não"
      />
      <label
        class="cursor-pointer capitalize text-slate-600 "
        for="não"
      >
        Não
      </label>
    </div>
  </div>
</DocumentFragment>
`;
